extension-impl/registry-zk/src/main/java/com/alipay/sofa/rpc/registry/zk/ZookeeperRegistryHelper.java

```diff
@@ -166,10 +167,7 @@ private static void addCommonAttrs(StringBuilder sb) {
         }
 
         for (ChildData childData : currentData) {
-            String url = childData.getPath().substring(providerPath.length() + 1); // 去掉头部
-            url = URLDecoder.decode(url, "UTF-8");
-            // byte[] data = childData.getData();
-            providerInfos.add(ProviderInfo.valueOf(url));
+            providerInfos.add(convertUrlToProvider(providerPath, childData));
```

Comment:
The `URLDecoder.decode(url, "UTF-8")` should not be deleted.

Reply:
The method `convertUrlToProvider`  has the `URLDecoder.decode(url, "UTF-8")`.